tuf/data/types.go

```diff
@@ -127,6 +128,68 @@ type FileMeta struct {
 	Custom json.RawMessage `json:"custom,omitempty"`
 }
 
+// CheckHashes verifies all the checksums specified by the "hashes" of the payload.
+func CheckHashes(payload []byte, hashes Hashes) error {
+	// k, v indicate the hash algorithm and the corresponding value
+	for k, v := range hashes {
+		switch k {
+		case notary.SHA256:
+			checksum := sha256.Sum256(payload)
+			if subtle.ConstantTimeCompare(checksum[:], v) == 0 {
+				return fmt.Errorf("%s checksum mismatched", k)
+			}
+		case notary.SHA512:
+			checksum := sha512.Sum512(payload)
+			if subtle.ConstantTimeCompare(checksum[:], v) == 0 {
+				return fmt.Errorf("%s checksum mismatched", k)
+			}
+		default:
+			return fmt.Errorf("unsupported hash algorithm")
```

Comment:
This shouldn't be an error. We validate all the hashes we understand and ignore any we don't. We must validate at least one hash, and all _recognized_ hashes must validate.